doc/dev/doc/serialization.md

```diff
@@ -0,0 +1,357 @@
+
+Here is an example of how you can construct a `HazelcastJsonValue` and put to the map:
+
+```csharp
+await map.PutAsync("item1", new HazelcastJsonValue("{ \"age\": 4 }"));
+await map.PutAsync("item2", new HazelcastJsonValue("{ \"age\": 20 }"));
+```
+
+You can query JSON objects in the cluster using the `Predicate`s of your choice. An example JSON query for querying the values whose age is less than 6 is shown below:
+
+```csharp
+    // Get the objects whose age is less than 6
+    var result = await map.GetValues(Predicates.IsLessThan("age", 6));
+
+    Console.WriteLine($"Retrieved {result.Count} values whose age is less than 6.");
+    Console.WriteLine($"Entry is: {result.First()}");
+```
+
+## Global Serialization
+
+The global serializer is identical to custom serializers from the implementation perspective. It is registered as a fallback serializer to handle all other objects if a serializer cannot be located for them. By default, the global serializer does not handle .NET Serializable instances. However, you can configure it to be responsible for those instances.
+
+A custom serializer should be registered for a specific class type. The global serializer will handle all class types if all the steps in searching for a serializer fail as described in Serialization Interface Types.
+
+**Use cases**
+
+- Third party serialization frameworks can be integrated using the global serializer.
+- For your custom objects, you can implement a single serializer to handle all of them.
+
+A sample global serializer that integrates with a third party serializer is shown below.
+
+```csharp
+public class GlobalSerializer : IStreamSerializer<object>
+{
+    public int TypeId => 20;
+
+    public void Write(IObjectDataOutput output, object obj)
+    {
+        output.write(MyFavoriteSerializer.serialize(obj))
+    }
+
+    public object Read(IObjectDataInput input)
+    {
+        return MyFavoriteSerializer.deserialize(input);
+    }
+}
+```
+
+You should register the global serializer in the configuration.
+
+**Programmatic Configuration:**
+```csharp
+var hazelcastOptions = new HazelcastOptionsBuilder().Build();
+hazelcastOptions.Serialization.GlobalSerializer.Creator =
+    () => new GlobalSerializer();
+```
+
+**Declarative Configuration:**
+```json
+{
+    "hazelcast": {
+        "serialization": {
+            "globalSerializer": {
+                "serializedTypeName": "Hazelcast.Examples.CustomSerializableType",
+                "overrideClrSerialization": true
```

Comment:
_HazelcastOptions.json_ in Tests has a bit different schema, not sure if casing matters:
```suggestion
                "typeName": "SampleNamespace.CustomSerializableType",
                "OverrideClrSerialization": true
```